merkletree.go

```diff
@@ -80,12 +93,13 @@ func lookUp(key string, str *SignedTreeRoot) (MerkleNode, []MerkleNode, error) {
 	}
 	switch nodePointer.(type) {
 	case *userLeafNode:
+		proof = append(proof, nodePointer.(*userLeafNode))
 		if nodePointer.(*userLeafNode).key == key {
-			proof = append(proof, nodePointer.(*userLeafNode))
 			return nodePointer.(*userLeafNode), proof, nil
-		} else {
-			// TODO: what should we return here?
 		}
+		// reached a different leaf with a matching prefix
+		// return nil and a auth path including the leaf node
+		return nil, proof, nil
```

Comment:
Ah I see now, ok. Thanks for clarifying! I think all fields included in the hash value of the leaf should be exported to allow the verifier to recompute the hash value. But I agree, this extension can be done in the future.